include/llbuild/Basic/Subprocess.h

```diff
@@ -224,6 +224,19 @@ namespace llbuild {
                                    const ProcessResult& result) = 0;
     };
 
+
+    struct ProcessAttributes {
+      /// If true, whether it is safe to attempt to SIGINT the process to cancel
+      /// it. If false, the process won't be interrupted during cancellation and
+      /// will be given a chance to complete (if it fails to complete it will
+      /// ultimately be sent a SIGKILL).
+      bool canSafelyInterrupt;
+
+      /// If set, the working directory to change into before spawning (only
+      /// supported on macOS)
```

Comment:
Has anyone investigated whether other platforms (Windows, Linux) have any such API?

At least, [CreateProcess](https://docs.microsoft.com/en-us/windows/desktop/api/processthreadsapi/nf-processthreadsapi-createprocessw) on Win32 appears to accept a working directory parameter (`lpCurrentDirectory`) which may do what we need. In `lib/llvm/Support/Windows/Program.inc` we're just passing NULL right now, which uses the calling process' working directory.

Reply:
Linux doesn't, as far as my research has shown. We can either fork, or we can use a "zygote" approach.

We don't really support Windows, so that is kind of a tangent.